sdk/trace/src/main/scala/org/typelevel/otel4s/sdk/trace/processor/SpanProcessor.scala

```diff
@@ -161,55 +201,64 @@ object SpanProcessor {
 
   private final class Noop[F[_]: Applicative] extends SpanProcessor[F] {
     val name: String = "SpanProcessor.Noop"
-
-    def isStartRequired: Boolean = false
-    def isEndRequired: Boolean = false
-
-    def onStart(parentCtx: Option[SpanContext], span: SpanRef[F]): F[Unit] =
-      Applicative[F].unit
-
-    def onEnd(span: SpanData): F[Unit] =
-      Applicative[F].unit
-
-    def forceFlush: F[Unit] =
-      Applicative[F].unit
+    val onStart: OnStart[F] = OnStart.noop
+    val onEnd: OnEnd[F] = OnEnd.noop
+    val forceFlush: F[Unit] = Applicative[F].unit
   }
 
   private final case class Multi[F[_]: MonadThrow: Parallel](
       processors: NonEmptyList[SpanProcessor[F]]
   ) extends SpanProcessor[F] {
-    private val startOnly: List[SpanProcessor[F]] =
-      processors.filter(_.isStartRequired)
-
-    private val endOnly: List[SpanProcessor[F]] =
-      processors.filter(_.isEndRequired)
-
     val name: String =
       s"SpanProcessor.Multi(${processors.map(_.name).mkString_(", ")})"
 
-    def isStartRequired: Boolean = startOnly.nonEmpty
-    def isEndRequired: Boolean = endOnly.nonEmpty
+    /** We use 'traverse' instead of 'parTraverse' due to:
+      *
+      * If multiple SpanProcessors are registered, their OnStart callbacks are invoked in the order they have been
+      * registered.
+      *
+      * Source: https://opentelemetry.io/docs/specs/otel/trace/sdk/#onstart
+      */
+    val onStart: OnStart[F] = {
+      val start = processors.filter { p =>
+        p.onStart match {
+          case _: OnStart.Noop[_] => false
+          case _                  => true
+        }
+      }
```

Comment:
possible cleaner (at least shorter) impl:

```suggestion
      val start = processors.filterNot(_.onStart.isInstanceOf[OnStart.Noop[_]])
```

can also be applied to `onEnd`, except that the diff decided to put a removed line in the middle so I can't add it as a suggestion

Reply:
Good point, thanks!